server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalExtendedStats.java

```diff
@@ -115,6 +120,11 @@ public double value(String name) {
         return super.value(name);
     }
 
+    @Override
+    public Iterable<String> valueNames() {
+        return metricNames;
```

Comment:
this is not related to this PR, but fixes an issue I found. I can move this into a separate PR if you prefer.